### PR TITLE
Add MenteeQuestionnaire model and validation tests

### PR DESCRIPTION
**Pull request by pratiktamang**
#### What's this PR do?

Add Mentee Questionnaire Model and basic tests.

Have not added the language attribute. This was also missing from mentor questionnaire, we're introducing a language model for that.

#### Model Attributes

| Attribute Name           | Type     | Question                                                      | null:false?  |
|--------------------------|----------|---------------------------------------------------------------|--------------|
| `respondent_id`          | binary   | User ID                                                       | -            |
| `name`                   | string   | What's your name?                                             | Required     |
| `work_url`               | string   | Are you working anywhere yet?                                  | -     |
| `writing_ruby`           | boolean  | Are you writing Ruby there?                                            | -     |
| `starting_point`         | string   | Where'd you get your start? Which bootcamp? Which university? Self-taught?                           | Required     |
| `underrepresented_group` | boolean  | Would you consider yourself in an underrepresented group?               | Required     |
| `twitter_handle`         | string   | Twitter                                                | -            |
| `github_handle`          | string   | GitHub                                                 | -            |
| `personal_site_url`      | string   | Do you have a personal site?                                             | -            |
| `previous_job`           | string   | What did you do before you became a programmer?                                                  | -            |
| `mentorship_goals`       | text     | What are you looking to get out of the mentoring?                      | Required     |
| `looking_for_career`     | boolean  | Are you looking for career development?                       | Required           |
| `looking_for_code`       | boolean  | Are you looking to improve your coding skills?                | Required            |

##### Background context
- MenteeQuestionnaire model outlined in [issue](https://github.com/goodscary/firstrubyfriend/issues/10)

#### Where should the reviewer start?
- `db/migrate/20231010180442_create_mentee_questionnaires.rb` for the schema changes
- `app/models/user.rb` for has_one 
- `app/models/mentee_questionnaire.rb` adds validations for null:false database constraints

Specific tricky bits of the code to focus on first.
n/a

#### How should this be manually tested?

n/a

#### Screenshots or Video

n/a

---

#### Migrations
- details above.

#### Additional deployment instructions

Anything specific we need to worry about when this hits production, do we need to worry about releases, migrations, downtime or significant changes to UI?

#### Additional ENV Vars

Any additional configuration required in development to run this code locally, **don't include actual secret values here**.

### CODE REVIEW

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class CreateMenteeQuestionnaires < ActiveRecord::Migration[7.0]
+  def change
+    create_table "mentee_questionnaires", id: :binary, force: :cascade do |t|
+      t.binary "respondent_id", limit: 16, null: false, foreign_key: true
+      t.string :name, null: false
+      t.string :country, null: false
+      t.string :city, null: false
+      t.string :work_url, null: false
+      t.boolean :writing_ruby, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:9]:
```suggestion
      t.boolean :currently_writing_ruby, null: false
```

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class CreateMenteeQuestionnaires < ActiveRecord::Migration[7.0]
+  def change
+    create_table "mentee_questionnaires", id: :binary, force: :cascade do |t|
+      t.binary "respondent_id", limit: 16, null: false, foreign_key: true
+      t.string :name, null: false
+      t.string :country, null: false
+      t.string :city, null: false
+      t.string :work_url, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:8]:
```suggestion
      t.string :work_url
```

They might not be working yet.

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class CreateMenteeQuestionnaires < ActiveRecord::Migration[7.0]
+  def change
+    create_table "mentee_questionnaires", id: :binary, force: :cascade do |t|
+      t.binary "respondent_id", limit: 16, null: false, foreign_key: true
+      t.string :name, null: false
+      t.string :country, null: false
+      t.string :city, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:7]: Don't need these here they're on user

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      t.string :work_url, null: false
+      t.boolean :writing_ruby, null: false
+      t.string :starting_point, null: false
+      t.boolean :underrepresented_group, null: false
+      t.string :twitter_handle
+      t.string :github_handle
+      t.string :personal_site_url
+      t.string :previous_job
+      t.text :mentorship_goals, null: false
+      t.boolean :looking_for_career, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:17]:
```suggestion
      t.boolean :looking_for_career_mentorship, null: false
```

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      t.boolean :looking_for_code, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:18]:
```suggestion
      t.boolean :looking_for_code_mentorship, null: false
```

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  def change
+    create_table "mentee_questionnaires", id: :binary, force: :cascade do |t|
+      t.binary "respondent_id", limit: 16, null: false, foreign_key: true
+      t.string :name, null: false
+      t.string :country, null: false
+      t.string :city, null: false
+      t.string :work_url, null: false
+      t.boolean :writing_ruby, null: false
+      t.string :starting_point, null: false
+      t.boolean :underrepresented_group, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:11]: Need a detail field to go with this one?

File: db/migrate/20231010180442_create_mentee_questionnaires.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
+class CreateMenteeQuestionnaires < ActiveRecord::Migration[7.0]
+  def change
+    create_table "mentee_questionnaires", id: :binary, force: :cascade do |t|
+      t.binary "respondent_id", limit: 16, null: false, foreign_key: true
+      t.string :name, null: false
+      t.string :country, null: false
+      t.string :city, null: false
+      t.string :work_url, null: false
+      t.boolean :writing_ruby, null: false
+      t.string :starting_point, null: false
```
Comment (andycroll) [inline db/migrate/20231010180442_create_mentee_questionnaires.rb:10]: Not totally sure what this question _is_

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 3 above)
Comment (andycroll) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:7]: I may have blithely added them to mentor questionnaire! Oops.

Comment (pratiktamang) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:7]: ah I overlooked here. Left out email but not country and city 😅.

Will get a separate PR with a migration to remove the rows from the table.

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 6 above)
Comment (pratiktamang) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:11]: 4 lines grouped together so unsure how to read this. 

Assuming its last line, if checked true then show a text field to capture details like this?
```
t.boolean :underrepresented_group, null: false
t.text :underrepresented_group_details
```

I also noticed we already added a `demographic_underrepresented_group` field on User table. So we only need to add a details field here?

https://github.com/goodscary/firstrubyfriend/blob/5b4afa7934d25c6be06d8d7bde24fbef63239ad6/db/schema.rb#L102

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 7 above)
Comment (pratiktamang) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:10]: Starting point is for: 

Where'd you get your start? Which bootcamp? Which university? Self-taught?

Comment (pratiktamang) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:10]: Updated the model attributes table in the PR description to precisely match with the current form questions

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 6 above)
Comment (andycroll) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:11]: Hmm. Good point... should this not be here and add a `demographic_underrepresented_group_details` text column on `users`.

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 7 above)
Comment (andycroll) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:10]: Maybe:

```suggestion
      t.string :where_started_coding, null: false
```

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 3 above)
Comment (andycroll) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb:7]: You can make the changes in here:

On your local branch:

1. `rails db:rollback`
2. Make the changes to this migration
3. `rails db:migrate`

Comment (pratiktamang) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb]: Sorry wasn't specific, I meant a separate migration to also remove it from the mentor_questionnaire model. I've removed it from this one in this PR.

File: db/migrate/20231010180442_create_mentee_questionnaires.rb (hunk 6 above)
Comment (pratiktamang) [reply db/migrate/20231010180442_create_mentee_questionnaires.rb]: PR to add this to Users table: https://github.com/goodscary/firstrubyfriend/pull/16

**Review by andycroll — APPROVED** [review]
All good!

**Comment by pratiktamang** [comment]
Perfect!

@andycroll I don't seem to have permissions to merge even though its approved. If you could please merge this and the other 2 and I'll start on the next PR after that. Thanks!


image

**Pull request merged**